library/alloc/src/collections/btree/borrow.rs

```diff
@@ -16,6 +16,9 @@ pub struct DormantMutRef<'a, T> {
     _marker: PhantomData<&'a mut T>,
 }
 
+unsafe impl<'a, T: Sync> Sync for DormantMutRef<'a, T> {}
+unsafe impl<'a, T: Send> Send for DormantMutRef<'a, T> {}
```

Comment:
Don't you want `&mut T` to be Send, not T?
```suggestion
unsafe impl<'a, T> Sync for DormantMutRef<'a, T> where &'a mut T: Sync {}
unsafe impl<'a, T> Send for DormantMutRef<'a, T> where &'a mut T: Send {}
```

Reply:
Is there any tangible difference? That implementation is already:

```rust
unsafe impl<T: Send + ?Sized> Send for &mut T {}
```

And I believe `Sync for &mut T` is auto-implemented for `T: Sync`.

Reply:
You could have `unsafe impl<T> Send for &mut T {}`, even if `T` is not Send.

Reply:
And in general this is conceptually a `&mut T` so it makes sense to use the same rules as &mut T for it.

Reply:
Yeah, I think they might match up in practice, but I didn't want to replicate impls when we could just use this.

Reply:
I guess `core` could pull that trick, but nobody else:

```rust
struct Foo(*const i32);

unsafe impl Send for &mut Foo {}
```

```
error[E0119]: conflicting implementations of trait `std::marker::Send` for type `&mut Foo`:
 --> src/lib.rs:3:1
  |
3 | unsafe impl Send for &mut Foo {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: conflicting implementation in crate `core`:
          - impl<T> std::marker::Send for &mut T
            where T: std::marker::Send, T: ?Sized;

error[E0321]: cross-crate traits with a default impl, like `std::marker::Send`, can only be implemented for a struct/enum type, not `&mut Foo`
 --> src/lib.rs:3:1
  |
3 | unsafe impl Send for &mut Foo {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't implement cross-crate trait with a default impl for non-struct/enum type
```